otherlibs/stdune/signal.ml

```diff
@@ -1,37 +1,125 @@
-let name =
-  let table =
-    let open Sys in
-    [ (sigabrt, "ABRT")
-    ; (sigalrm, "ALRM")
-    ; (sigfpe, "FPE")
-    ; (sighup, "HUP")
-    ; (sigill, "ILL")
-    ; (sigint, "INT")
-    ; (sigkill, "KILL")
-    ; (sigpipe, "PIPE")
-    ; (sigquit, "QUIT")
-    ; (sigsegv, "SEGV")
-    ; (sigterm, "TERM")
-    ; (sigusr1, "USR1")
-    ; (sigusr2, "USR2")
-    ; (sigchld, "CHLD")
-    ; (sigcont, "CONT")
-    ; (sigstop, "STOP")
-    ; (sigtstp, "TSTP")
-    ; (sigttin, "TTIN")
-    ; (sigttou, "TTOU")
-    ; (sigvtalrm, "VTALRM")
-    ; (sigprof, "PROF")
-    ; (sigbus, "BUS")
-    ; (sigpoll, "POLL")
-    ; (sigsys, "SYS")
-    ; (sigtrap, "TRAP")
-    ; (sigurg, "URG")
-    ; (sigxcpu, "XCPU")
-    ; (sigxfsz, "XFSZ")
-    ]
-  in
-  fun n ->
-    match List.assoc table n with
-    | None -> Printf.sprintf "%d\n" n
-    | Some s -> s
+type t =
+  | Int
+  | Term
+  | Abrt
+  | Alrm
+  | Fpe
+  | Hup
+  | Ill
+  | Kill
+  | Pipe
+  | Quit
+  | Segv
+  | Usr1
+  | Usr2
+  | Chld
+  | Cont
+  | Stop
+  | Tstp
+  | Ttin
+  | Ttou
+  | Vtalrm
+  | Prof
+  | Bus
+  | Poll
+  | Sys
+  | Trap
+  | Urg
+  | Xcpu
+  | Xfsz
+  | Unknown of int
+
+let all =
```

Comment:
How about getting rid of this table too? We can do this as follows:

* Implement `of_int` using pattern matching.

* Implement `to_int` by creating an `Int.Map.t` and looking up in it.

Traversing lists on each call just doesn't seem right to me. (Although I may be overthinking this.)

Reply:
This happens so rarely that the performance gained negligible, but I can make it a hash table lookup both ways if you want.  At least we won't need to duplicate much code if we add a new signal.

Reply:
I don't think any duplication will be needed with the above idea?

But we can also keep the current implementation if you prefer -- just please add some comments saying that it's currently pretty slow.

Reply:
If we implement `of_int` using pattern matching and `to_int` with a map lookup, we will need to modify two places when we add a new signal:

* The list `(int * t) list`which enumerates all the known signal
* The pattern match which converts signals to integers.

If we start with a list and generate two tables from it, we only need to update this single list.

Reply:
I thought we don't need the list of known signals but I guess I was wrong.